src/control/client/mocks.go

```diff
@@ -27,8 +27,52 @@ import (
 	"fmt"
 
 	pb "github.com/daos-stack/daos/src/control/common/proto/mgmt"
+	"google.golang.org/grpc/connectivity"
 )
 
+// implement mock/stub behaviour for Control
+type mockControl struct {
+	address   string
+	connState connectivity.State
+	features  []*pb.Feature
+	ctrlrs    NvmeControllers
+	modules   ScmModules
+}
+
+func (m *mockControl) connect(addr string) error {
+	m.address = addr
+	return nil
+}
+func (m *mockControl) disconnect() error { return nil }
+func (m *mockControl) connected() (connectivity.State, bool) {
+	return m.connState, checkState(m.connState)
+}
+func (m *mockControl) getAddress() string { return m.address }
+func (m *mockControl) listAllFeatures() (FeatureMap, error) {
+	fm := make(FeatureMap)
+	for _, f := range m.features {
+		fm[f.Fname.Name] = fmt.Sprintf(
+			"category %s, %s", f.Category.Category, f.Description)
+	}
+	return fm, nil
+}
+func (m *mockControl) listNvmeCtrlrs() (NvmeControllers, error) {
+	return m.ctrlrs, nil
+}
+func (m *mockControl) listScmModules() (ScmModules, error) {
+	return m.modules, nil
+}
+func (m *mockControl) killRank(uuid string, rank uint32) error {
+	return nil
```

Comment:
No change needed for this patch, but it would be interesting to unit test the case where an error is returned.